lua/wikis/commons/Widget/MainPage/NavigationCard.lua

```diff
@@ -29,23 +30,51 @@ local NavigationCard = Class.new(Widget)
 ---@return Widget
 function NavigationCard:render()
 	local count = self.props.count
+
+	-- Determine which type of content to show
+	local hasIcon = self.props.iconClasses ~= nil and self.props.iconClasses ~= ''
+	local contentDiv
+
+	if Logic.isNotEmpty(self.props.iconClasses) then
+		-- Icon rendering
+		contentDiv = HtmlWidgets.Div{
+			classes = {'navigation-card__icon'},
+			children = HtmlWidgets.I{
+				classes = self:processIconClasses(self.props.iconClasses),
+			}
+		}
```

Comment:
Icon rendering should be done with commons/icon preferably with commoms/widget/image/fontawesome

Reply:
But that's harcoding every icon. Don't we want to give freedom in choosing an icon?

Reply:
@mbergen could I have a second pair of eyes? `IconData[self.props.iconName]` should be correct right?